src/panels/config/areas/ha-config-area-page.ts

```diff
@@ -136,10 +144,60 @@ class HaConfigAreaPage extends LitElement {
       this.entities
     );
 
-    const grouped = groupBy(entities, (entity) =>
+    // Pre-compute the entity and device names, so we can sort by them
+    if (devices) {
+      devices.forEach((entry) => {
+        entry.name = computeEntityRegistryName(this.hass, entry);
+      });
+      sortDeviceRegistryByName(devices);
+    }
+    if (entities) {
+      entities.forEach((entry) => {
+        entry.name = computeEntityRegistryName(this.hass, entry);
+      });
+      sortEntityRegistryByName(entities);
+    }
+
+    // Group entities by domain
+    const groupedEntities = groupBy(entities, (entity) =>
       computeDomain(entity.entity_id)
     );
 
+    // Pre-compute the name also for the grouped and related entities so we can sort by them
+    let groupedAutomations: NameAndEntity<AutomationEntity>[] = [];
+    let groupedScenes: NameAndEntity<SceneEntity>[] = [];
+    let groupedScripts: NameAndEntity<ScriptEntity>[] = [];
+    let relatedAutomations: NameAndEntity<AutomationEntity>[] = [];
+    let relatedScenes: NameAndEntity<SceneEntity>[] = [];
+    let relatedScripts: NameAndEntity<ScriptEntity>[] = [];
+
+    if (isComponentLoaded(this.hass, "automation")) {
+      const data = this._prepareEntities<AutomationEntity>(
+        groupedEntities.automation,
+        this._related?.automation
+      );
+      groupedAutomations = data.groupedEntities;
+      relatedAutomations = data.relatedEntities;
```

Comment:
```suggestion
      const {groupedEntities: groupedAutomations, relatedEntities: relatedAutomations} = this._prepareEntities<AutomationEntity>(
        groupedEntities.automation,
        this._related?.automation
      );
```

Reply:
That will not work, since the destructured elements are already declared in the upper scope (they need to be there since the rendering further down below needs them to be defined, regardless of what happens with the `_prepareEntities()`).

Reply:
Then do:
```ts
({groupedEntities: groupedAutomations, relatedEntities: relatedAutomations} = this._prepareEntities<AutomationEntity>(
        groupedEntities.automation,
        this._related?.automation
      ));
```

Reply:
That is what I wanted to do, but it never worked...because I had added the closing round bracket before the `=`.  Did not occur to me that the whole statement needs to be bracketed for the unpacking to work 😆 .